### DISCUSSION TITLE
[java] : detecting function within a declared function that has particular name

### DISCUSSION THREAD

**Discussion by ahmed-farid-dev** [github/codeql#10474] [Q&A]
Hi,  if you can help me to detect a function inside a declared function that has a particular name like this:
```
private boolean login(String pwd, HttpServletRequest request) {
    String password = request.getParameter("password");
    return password.equals(pwd);        
}
```
I wanna detect equals() function and in the same time matching  the name of the declared function in this case login().

**Comment by atorralba** [comment]
You can use `getEnclosingCallable`, e.g. (untested code):

```ql
predicate test() {
  exists(MethodAccess equals, Method login |
    equals.getMethod().hasName("equals") and
    login.hasName("login") and
    equals.getEnclosingCallable() = login
  )
}
```

Of course, feel free to further restrict `login` and `equals` based on their declaring types, qualifiers, or whatever else you need.
